ledger/test-common/BUILD.bazel

```diff
@@ -86,19 +86,39 @@ alias(
     for target in lf_version_configuration_versions
 ]
 
-# Correspond to the directories under src/test/lib/daml
-test_names = [
+# Carbon tests helps to test upgrades when packages are dependent on each other
+carbon_test_names = [
+    "carbonv1",
+    "carbonv2",  #carbonv2 depends on carbon v1
+]
+
+general_test_names = [
     "benchtool",
     "model",
     "semantic",
     "performance",
     "package_management",
 ]
 
+# Correspond to the directories under src/test/lib/daml
+test_names = {
+    "1.dev": general_test_names + carbon_test_names,
+    "1.8": general_test_names,
+    "1.14": general_test_names,
+}
+
 da_scala_dar_resources_library(
     add_maven_tag = True,
     daml_dir_names = test_names,
     daml_root_dir = "src/main/daml",
+    dar_dict = {
```

Comment:
I'm not sure you actually need `dar_dict` here. That is intended for the rules that have a daml.yaml file checked in and we need to patch that (mainly for docs, where this stuff matters). The rules here generate their own daml.yaml so you should be able to just specify `data_dependencies` as a list and put that in the generated daml.yaml.

Reply:
My understanding is that `dar_dict` here actually copies the output of V1 dar build to V2 so you can actually refer to the definitions in V2 to V1. When I removed this line I've got failure of the build:

```
damlc-compile-only: path/to/carbon-1.0.0.dar: openBinaryFile: does not exist (No such file or directory)
```

Or do you mean I have to change the macro further to imply this `dar_dict` from `data_dependencies` ?

Reply:
I think you're right that `_daml_build` might still want you to pass a dar dict. But I think I'd at least push the generation of that dict into `daml_compile`s impl instead of having the caller specify this.

Reply:
But not sure you can actually do it. `data_dependencies` specifies for a package where to get the dependency locally (i.e. v2 of the package needs a file from `path/to/carbon-1.0.0.dar`), but it does not know which package it needs (v1 in our case). 

Otherwise I could pass another dictionary with the information that v2 depends on v1 and imply both: data_dictionary and dar_dict. It would make `da_scala_dar_resources_library` slightly more simpler. Is this what you actually wanted?

Reply:
The UX I'd aim for is
```
data_dependencies = {
  "cabonv2-tests-1.dev": ["//ledger/test-common:carbonv1-tests-1.dev.build"],
},
```

So you specify directly the list of dars you depend on rather than doing that substitution thingy.

Reply:
and would you just modify `da_scala_dar_resources_library` for that or it would go even deeper into `daml_compile` ?

Reply:
up to daml_compile. there you then just accept a flat list of `data_dependencies`.